crates/mirror-mirror/src/type_info/mod.rs

```diff
@@ -641,6 +641,10 @@ impl<'a> StructType<'a> {
         })
     }
 
+    pub fn fields_len(self) -> usize {
```

Comment:
(light suggestion, take or dismiss as you wish) Here and everywhere in this PR, I think that `num_fields` for "number of fields" would be a slightly more common function name.

Reply:
Hm yeah maybe 🤔 I think I'll stick with `fields_len` for now because then it matches [`Struct::fields_len`](https://docs.rs/mirror-mirror/latest/mirror_mirror/trait.Struct.html#tymethod.fields_len).